### PR TITLE
Issue/47 implement date time form fields

### PR DESCRIPTION
**Pull request by detautama**
Resolves https://github.com/oddbit/tanam/issues/47

Date time

image

### CODE REVIEW

File: angular/projects/admin/src/app/admin/document/document-form/document-form.component.html [hunk 1]
```diff
@@ -56,6 +56,31 @@
...
+                </mat-form-field>
+
+
+                <!-- Time Picker -->
+                <mat-form-field *ngSwitchCase="'time'">
+                  <mat-placeholder>{{field.title}}</mat-placeholder>
+                  <mat-datetimepicker-toggle [for]="timePicker" matSuffix></mat-datetimepicker-toggle>
+                  <mat-datetimepicker #timePicker type="time" openOnFocus="true"></mat-datetimepicker>
+                  <input matInput formControlName="{{field.key}}" [matDatetimepicker]="timePicker" autocomplete="false">
+                </mat-form-field>
```
Comment (DennisAlund) [inline angular/projects/admin/src/app/admin/document/document-form/document-form.component.html]: We can remove the picker that only selects "time" for now. It is unclear when a time without a date would be.

File: angular/projects/admin/src/app/services/document.service.ts [hunk 2]
```diff
@@ -51,6 +52,17 @@ export class DocumentService {
       throw new Error('Document ID must be provided as an attribute when updating an document.');
     }
 
+    // Convert moment data to firestore timestamp
+    const data = document.data;
+    for (const key in data) {
+      if (moment.isMoment(data[key])) {
+        const toDate = moment(data[key]).toDate();
+        const toTimestamp = firebase.firestore.Timestamp.fromDate(toDate);
```
Comment (DennisAlund) [inline angular/projects/admin/src/app/services/document.service.ts]: Move this handling closer to the form element that provides a value. Optimally the date-time picker returns a Timestamp. But if it is not possible, then the form should be the only one knowing about the `moment` data type. 
Let the form or the form component change the value to `Timestamp` before passing to the service.

File: angular/projects/admin/src/app/admin/document/document-form/document-form.component.html (hunk 1 above)
Comment (DennisAlund) [reply angular/projects/admin/src/app/admin/document/document-form/document-form.component.html]: That means that you can remove this line too: https://github.com/oddbit/tanam/blob/0e4996fc5c5828aafed7e2956c084ebad034d0b3/models/document-type.models.ts#L7

Comment (detautama) [reply angular/projects/admin/src/app/admin/document/document-form/document-form.component.html]: Resolve on this commit https://github.com/oddbit/tanam/pull/56/commits/15d400fcc5d319e6335e4dea8f6012c9713a2ed7

File: angular/projects/admin/src/app/services/document.service.ts (hunk 2 above)
Comment (detautama) [reply angular/projects/admin/src/app/services/document.service.ts]: Resolve on this commit https://github.com/oddbit/tanam/pull/56/commits/f103dd165b0bdeafa258cf7b7c488043900b7d50

File: angular/projects/admin/src/app/admin/form/date-time/date-time.component.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+
+  stateChanges = new Subject<void>();
+  controlType = 'date-time';
+  shouldLabelFloat = false;
+
+  required: boolean;
+  errorState: boolean;
+  autofilled?: boolean;
+  editorData: any;
```
Comment (DennisAlund) [inline angular/projects/admin/src/app/admin/form/date-time/date-time.component.ts]: Please change the variable name to `timestamp` and the type should be `firestore.Timestamp`

File: angular/projects/admin/src/app/admin/form/date-time/date-time.component.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+  setDescribedByIds(ids: string[]) {
+    this.describedBy = ids.join(' ');
+  }
+
+  onContainerClick(event: MouseEvent): void {
+    this._onTouchedCallback();
+  }
+
+  writeValue(text: firestore.Timestamp): void {
```
Comment (DennisAlund) [inline angular/projects/admin/src/app/admin/form/date-time/date-time.component.ts]: Parameter name should be `timestamp` not `text`

File: angular/projects/admin/src/app/admin/form/date-time/date-time.component.ts (hunk 3 above)
Comment (detautama) [reply angular/projects/admin/src/app/admin/form/date-time/date-time.component.ts]: Resolve on this commit https://github.com/oddbit/tanam/pull/56/commits/afe64bb0f806d9ac25dc11b55c4a023e3c778a9d

File: angular/projects/admin/src/app/admin/form/date-time/date-time.component.ts (hunk 4 above)
Comment (detautama) [reply angular/projects/admin/src/app/admin/form/date-time/date-time.component.ts]: Resolve on this commit https://github.com/oddbit/tanam/pull/56/commits/afe64bb0f806d9ac25dc11b55c4a023e3c778a9d

**Pull request merged**